overlord/snapstate/backend/setup.go

```diff
@@ -356,16 +361,20 @@ func compsMountPoints(comps []*snap.ComponentSideInfo, kSnapName string, ksnapRe
 		}
 	}
 
-	// Add folders in SNAP_DATA if $SNAP_DATA/{modules,firmware} exist.
-	dataDir := snap.DataDir(kSnapName, ksnapRev)
-	if dirHasDrivers(dataDir) {
-		logger.Noticef("setup: modules for %s", dataDir)
-		compsMntPts = append(compsMntPts, kernel.ModulesCompMountPoints{
-			LinkName: kSnapName + "_dyn",
-			MountPoints: kernel.MountPoints{
-				Current: dataDir,
-				Target:  dataDir,
-			}})
+	// These are the only valid values at the moment - if
+	// kernel.InfoFromKernelYaml changes this needs to be changed too.
+	if dynModules == "$SNAP_DATA" || dynModules == "${SNAP_DATA}" {
```

Comment:
Would it be better to resolve `dynModules` at the callsite of this function? So `dynModules` will actually point to a path, rather than either a path or a variable?

Reply:
made a slightly different suggestion